core/src/main/scala/org/apache/spark/crypto/CryptoStreamUtils.scala

```diff
@@ -0,0 +1,107 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.spark.crypto
+
+import java.io.{IOException, InputStream, OutputStream}
+import java.nio.ByteBuffer
+import java.util.Properties
+
+import com.intel.chimera.ConfigurationKeys._
+import com.intel.chimera.codec.CipherSuite
+import com.intel.chimera.codec.CryptoCodec
+import com.intel.chimera._
+import com.intel.chimera.random.OpensslSecureRandom
+
+import sun.misc.Cleaner
+import sun.nio.ch.DirectBuffer
+
+import org.apache.spark.crypto.CommonConfigurationKeys._
+import org.apache.spark.deploy.SparkHadoopUtil
+import org.apache.spark.SparkConf
+
+/**
+ * A util class for CryptoInputStream and  CryptoOutputStream
+ */
+private[spark] object CryptoStreamUtils {
+  val MIN_BUFFER_SIZE = 512
+  val IV_LENGTH = 16
+
+  /** Forcibly free the direct buffer. */
+  def freeDB(buffer: ByteBuffer): Unit = {
+    if (buffer.isInstanceOf[DirectBuffer]) {
+      val bufferCleaner: Cleaner = (buffer.asInstanceOf[DirectBuffer]).cleaner()
+      bufferCleaner.clean()
+    }
+  }
+
+  /** Read crypto buffer size */
+  private[this] def getBufferSize(sparkConf: SparkConf): Int = {
+    sparkConf.getInt(SPARK_SHUFFLE_CRYPTO_BUFFER_SIZE_KEY, CommonConfigurationKeys
+        .SPARK_SHUFFLE_CRYPTO_BUFFER_SIZE_DEFAULT)
+  }
+
+  /** AES/CTR/NoPadding is required */
+  def checkCodec(codec: CryptoCodec): Unit = {
+    if (codec.getCipherSuite != CipherSuite.AES_CTR_NOPADDING) {
+      throw new RuntimeException("AES/CTR/NoPadding is required")
+    }
+  }
+
+  def createCryptoOutputStream(os: OutputStream, sparkConf: SparkConf): CryptoOutputStream = {
```

Comment:
What do you thing about moving this method and `createCryptoInputStream` into the library itself? It seems like it would be more generally useful there. It would have to take the key as an argument (so you can still have this wrapper here, it would just be a lot simpler).